Refactor ItemServiceImpl: rename findByMetadataField to findArchivedByMetadataField to improve clarity and remove unnecessary validation of the metadata schema.

jesielviana opened the pull request:
## References
* Fixes #11513

## Description
Renamed the `findByMetadataField` method to `findArchivedByMetadataField` for improved clarity, as this method only returns archived items.

## Instructions for Reviewers
List of changes in this PR:
* Removed the old `findArchivedByMetadataField` method.
* Renamed `findByMetadataField` to `findArchivedByMetadataField` to make its purpose clearer.
* Removed the unnecessary validation of the metadata schema, since the metadata field lookup already includes the schema check — this avoids an extra database call.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome).
However, reviewers may request that you complete any actions in this list if you have not done so. If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [ ] My PR is **created against the `main` branch** of code (unless it is a backport or is fixing an issue specific to an older branch).
- [ ] My PR is **small in size** (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [ ] My PR **passes Checkstyle** validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [ ] My PR **includes Javadoc** for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [ ] My PR **passes all tests and includes new/updated Unit or Integration Tests** based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] My PR **includes details on how to test it**. I've provided clear instructions to reviewers on how to successfully test this fix or feature.
- [ ] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [ ] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [ ] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

jesielviana committed:
rename findByMetadataField to findArchivedByMetadataField to improve clarity

jesielviana committed:
remove unnecessary validation of the metadata schema

MW3000 reviewed (changes requested):
I read through all the changes. Looks all straightforward and correct ✅

You could add one missing exception in a JavaDoc comment. I left a comment.

There appeared two merge conflict with recent changes. So, you could address those and then it can be merged.

jesielviana committed:
add IOException in the java doc

jesielviana committed:
Merge branch 'main' into refactor-itemservice

MW3000 reviewed (approved):
Reviewed a second time after fix of comment and solution of merge conflicts.
Looks straight-forward. Can be merged like this ✅

tdonohue commented on dspace-api/src/main/java/org/dspace/content/service/ItemService.java:
```diff
@@ -738,11 +722,13 @@ Iterator<Item> findArchivedByMetadataField(Context context, String schema,
      */
     Iterator<Item> findArchivedByMetadataField(Context context, String metadataField, String value)
-            throws SQLException, AuthorizeException;
+            throws SQLException, AuthorizeException, IOException;
```
@jesielviana : As Copilot rightly noted here, it's unclear why you added `IOException` to the "throws"?  It doesn't look like this method throws an `IOException` at all, and it's required you to update other classes with `throws IOException`.  Could you explain why this change was necessary, or remove the change?

tdonohue reviewed (commented):
@jesielviana : This has PR merge conflicts.  Could you fix those?   Also, I have a question inline below as it's unclear to me why you had to add `IOException` to the list of exceptions thrown...that seems to have made this PR larger because you've had to update other areas of the code to now handle that `IOException`

jesielviana committed:
Merge branch 'main' into refactor-itemservice

jesielviana committed:
Merge branch 'main' into refactor-itemservice and refactoring changes

# Conflicts:
#	dspace-api/src/main/java/org/dspace/identifier/EZIDIdentifierProvider.java

jesielviana committed:
Merge remote-tracking branch 'origin/refactor-itemservice' into refactor-itemservice

# Conflicts:
#	dspace-api/src/main/java/org/dspace/identifier/EZIDIdentifierProvider.java

jesielviana committed:
Fix compile error

jesielviana commented:
@tdonohue, as shown in the first commit, I removed the old `findArchivedByMetadataField` method and renamed `findByMetadataField` to `findArchivedByMetadataField`. The `IOException` was already declared in the original `findByMetadataField` method, so it was not newly introduced—it was simply carried over during the renaming.

I have now removed it, as I agree that it is not needed. Thank you for your report.

tdonohue commented on dspace-server-webapp/src/main/java/org/dspace/app/rest/login/impl/ResearcherProfileAutomaticClaim.java:
```diff
@@ -11,6 +11,7 @@
 import static org.apache.commons.lang3.StringUtils.isBlank;
 import static org.dspace.content.authority.Choices.CF_ACCEPTED;
 
+import java.io.IOException;
```
@jesielviana : Now that you've removed the unnecessary `IOException`, you may be able to remove this import in this class as well.  In fact, I think you can undo all the changes in this class as you only made changes related to `IOException` here.

jesielviana committed:
remove the unnecessary IOException

tdonohue reviewed (approved):
👍 Thanks @jesielviana !  Sorry for the delays in getting back to this. But, this also looks good to me.
